### PR TITLE
[NUI] Open classes of components

### PR DESCRIPTION
**Pull request by AdunFang**
### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI.Components/Controls/DropDown.cs [hunk 1]
```diff
@@ -530,77 +505,19 @@ public ListOrientation ListRelativeOrientation
...
-        public int ListBottomMargin
-        {
-            get
-            {
-                return (int)dropDownAttributes.ListMargin.W;
-            }
-            set
-            {
-                dropDownAttributes.ListMargin.W = value;
+                dropDownAttributes.ListMargin = new Extents(value.Start, value.End, value.Top, value.Bottom);
```
Comment (zhouleonlei) [inline src/Tizen.NUI.Components/Controls/DropDown.cs]: OK

**Comment by m-macias** [comment]
The removal of EditorBrowsableState.Never attribute requires an ACR on Jira. Please don't merge it.

**Comment by dongsug-song** [comment]
@AdunFang 
@elishateng 
Hello, 
"ACR Required" tag was attached, please check if some hidden API is changed to be opened as public.
you need to do TCSACR and update the ACR link here.
before ACR is accepted, we can not merge the patch, as you know.
thank you.

**Comment by hyunjushin** [comment]
CheckBox is inherited from SelectButton. But SelectButton looks also hidden.

**Comment by dongsug-song** [comment]
@AdunFang 
Hello,
could you check the updated review comments and answer?

File: src/Tizen.NUI.Components/Attributes/DropDownAttributes.cs [hunk 2]
```diff
@@ -306,12 +306,12 @@ public ImageAttributes CheckImageAttributes
         }
 
         /// <summary>
-        /// Right space from check image.
+        /// Gap of Check image to boundary.
```
Comment (agnelovaz) [inline src/Tizen.NUI.Components/Attributes/DropDownAttributes.cs]: Is the boundary the border of the Control? If so then should it state which axis?  Could this not be done with Padding?

File: src/Tizen.NUI.Components/Controls/CheckBoxGroup.cs [hunk 3]
```diff
@@ -153,7 +145,6 @@ public bool IsCheckedAll()
         /// <param name="state">The boolean state of the check box</param>
         /// <since_tizen> 6 </since_tizen>
         /// This will be public opened in tizen_5.5 after ACR done. Before ACR, need to be hidden as inhouse API.
-        [EditorBrowsable(EditorBrowsableState.Never)]
         public void CheckingAll(bool state)
```
Comment (agnelovaz) [inline src/Tizen.NUI.Components/Controls/CheckBoxGroup.cs]: public void CheckAll(bool state)

File: src/Tizen.NUI.Components/Controls/CheckBoxGroup.cs [hunk 4]
```diff
@@ -31,15 +31,13 @@ namespace Tizen.NUI.Components
     /// </code>
     /// <since_tizen> 6 </since_tizen>
     /// This will be public opened in tizen_5.5 after ACR done. Before ACR, need to be hidden as inhouse API.
-    [EditorBrowsable(EditorBrowsableState.Never)]
     public class CheckBoxGroup : SelectGroup
```
Comment (agnelovaz) [inline src/Tizen.NUI.Components/Controls/CheckBoxGroup.cs]: Do we need to make this Public now?  Wondering if a SelectGroup is really needed, instead Buttons could be grouped by the parent container they are added too.   Currently grouped buttons need adding to a parent View and also added to a SelectGroup. I'm assuming that grouping buttons would need a visual (a box) indication not just conceptual group.

File: src/Tizen.NUI.Components/Controls/InputField.cs [hunk 5]
```diff
@@ -537,7 +517,6 @@ protected virtual bool OnTextFieldKeyEvent(object source, KeyEventArgs e)
         /// <param name="h">Input Field' height.</param>
         /// <since_tizen> 6 </since_tizen>
         /// This will be public opened in tizen_5.5 after ACR done. Before ACR, need to be hidden as inhouse API.
-        [EditorBrowsable(EditorBrowsableState.Never)]
         protected void SetTextFieldSize2D(int w, int h)
```
Comment (agnelovaz) [inline src/Tizen.NUI.Components/Controls/InputField.cs]: Should this have been renamed to remove 2D?

**Comment by hyunjushin** [comment]
please reopen after refactoring this
"- Applying BinableObject Pattern to bind View and Attributes
Move Background and Shadow properties to View from Components
In Button, provide Icon and Text property as ImageAttributes and TextAttributes repectively.
"

**Pull request closed without merging**